ansible_events_ui/api/project.py

```diff
@@ -125,18 +139,24 @@ async def update_project(
     query = sa.select(projects).where(projects.c.id == project_id)
     stored_project = (await db.execute(query)).first()
     if not stored_project:
-        raise HTTPException(status_code=404, detail="Project not found")
+        raise HTTPException(status_code=NOT_FOUND, detail="Project not found")
+
+    for key, val in project:
+        if not val:
+            project.__setattr__(key, stored_project[key])
```

Comment:
1. In general you should avoid calling dunder methods directly without a reason. There are `getattr` \ `setattr` \ `hasattr` \ `delattr` buitin functions for that purpose.
2. Can you please explain why this code is needed?

Reply:
Originally I was going to add this code to make sure we were not accidentally pushing NULL values into the DB on updates, however after some thought I don't believe this piece will be necessary.